### PR TITLE
rails 6.1 upgrade

### PR DESCRIPTION
**Pull request by yannski**

### CODE REVIEW

File: app/models/job_application.rb [hunk 1]
```diff
@@ -169,17 +169,20 @@ def all_available_file_types
...
 
+  # Return two arrays
+  # First with JobApplicationFile already existing
+  # Second with JobApplicationFileType
   def files_to_be_provided
     JobApplicationFileType.all.each_with_object([[], []]) do |file_type, memo|
-      existing = job_application_files.detect do |x|
-        x.job_application_file_type_id == file_type.id
-      end
+      existing_file = job_application_files.find_by(job_application_file_type: file_type)
```
Comment (yannski) [inline app/models/job_application.rb:177]: will that make an SQL query for each call? That's what I wanted to avoid in the first place :)

Comment (btrd) [reply app/models/job_application.rb:177]: That's right ! I will rollback this change sorry

**Review by btrd — APPROVED** [review]

**Comment by btrd** [comment]
@yannski If it's ok for you I think we can merge

File: db/seeds.rb [hunk 2]
```diff
@@ -272,12 +272,6 @@
   analyses contextuelles et anticiper les modalités de confinement, de remédiation et de
   reconstruction des technologies
   les plus impactantes pour le ministère de la Défense.
-
```
Comment (yannski) [inline db/seeds.rb:276]: why?

File: app/models/job_offer_actor.rb [hunk 3]
```diff
@@ -8,22 +8,16 @@ class JobOfferActor < ApplicationRecord
...
-    existing ||= Administrator.where(email: attributes[:email]).first
-    if existing
-      if has_destroy_flag?(attributes)
-        mark_for_destruction
-      else
-        self.administrator = existing
-      end
-    elsif has_destroy_flag?(attributes)
-      mark_for_destruction
+    mark_for_destruction if has_destroy_flag?(attributes)
```
Comment (yannski) [inline app/models/job_offer_actor.rb:11]: did this method had a test suite? if not, can you add one?

**Comment by yannski** [comment]
@btrd can you rewrite the git history to remove the git merge commit?

**Comment by yannski** [comment]
@btrd can you verify the activestorage related migrations will work with our current uuid scheme?

File: db/seeds.rb (hunk 2 above)
Comment (btrd) [reply db/seeds.rb:276]: It's more than 1000 char https://github.com/betagouv/civilsdeladefense/issues/441

File: spec/models/job_offer_spec.rb [hunk 4]
```diff
@@ -152,10 +152,8 @@
     brh = create(:administrator, role: nil, employer: employer)
     other_admin = create(:administrator, role: nil, employer: employer)
 
-    job_offer = create(:job_offer) do |obj|
-      actor_attrs = attributes_for(:job_offer_actor, role: 'brh', administrator: brh)
-      obj.job_offer_actors.create(actor_attrs)
-    end
```
Comment (btrd) [inline spec/models/job_offer_spec.rb:158]: It wasn't working anymore with the change in job_offer_actor factory

File: app/models/job_offer_actor.rb (hunk 3 above)
Comment (btrd) [reply app/models/job_offer_actor.rb:11]: https://github.com/betagouv/civilsdeladefense/pull/771/commits/a9be8c391f880bc86f89f8c480ab1833cd877688#diff-300e259f4866f0ba878a0ab70211af9453df4a87d60eba34ce5eaabf4774041cL155-L158

Done

**Pull request merged**